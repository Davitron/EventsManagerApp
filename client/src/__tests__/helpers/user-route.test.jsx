import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import UserRoute from '../../helpers/user-route';
import CenterEvent from '../../components/event/CenterEvent';

let mountedComponent;
let wrapper;
let props;

/**
 * @description Initialise the component
 *
 * @returns {object} mountedComponent - Mounte
 */
const getComponent = () => {
  if (!mountedComponent) {
    props = {
      path: '/pending-events',
      component: CenterEvent,
      redirectPath: '/login',
      exact: true
    };
    mountedComponent = shallow(<UserRoute {...props} />);
  }
  return mountedComponent;
};

describe('Event Component', () => {
  beforeAll(() => {
    wrapper = getComponent();
  });


  it('component successfully rendered', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
});
