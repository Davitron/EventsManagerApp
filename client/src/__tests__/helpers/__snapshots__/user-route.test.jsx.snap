// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserRoute
    component={[Function]}
    exact={true}
    path="/pending-events"
    redirectPath="/login"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "exact": true,
      "path": "/pending-events",
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "exact": true,
        "path": "/pending-events",
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
